Mark

Mackoyokcam opened the pull request.

Mackoyokcam committed:
initial scaffolding

Mackoyokcam committed:
Queue works

Mackoyokcam committed:
Merge pull request #1 from Mackoyokcam/queues

Queue works

vladimirsan commented on lab-mark/README.md:
```diff
@@ -0,0 +1,10 @@
+Description of exported values of each module defined in lib/ directory, along with airity and expected input/return values.
+### queue.js
+Exports a factory function that creates a queue using a linked list architecture.
+* ##### Methods
```
Add some more information about what happens in special cases. e.g. what happens if we do myQueue.enqueue(undefined);

vladimirsan commented on lab-mark/__tests__/queue-test.js:
```diff
@@ -0,0 +1,18 @@
+
+const getQueue = () => Queue();
+
+describe('Queue', () => {
```
Try adding tests a few test with multiple enqueues / dequeues in sequence

vladimirsan commented on lab-mark/model/queue.js:
```diff
@@ -0,0 +1,30 @@
+        last = sll(value);
+      } else {
+        let lastIn = sll(value);
+        if(!first.next)
```
Why do you need this if?

Mackoyokcam committed:
Updated README.md and queue-test.js

README.md now explains special cases
queue-test.js now tests for enqueue and dequeues in different sequences.

Mackoyokcam committed:
Merge pull request #2 from Mackoyokcam/queues

Updated README.md and queue-test.js
